runtime/hydradx/src/evm/mod.rs

```diff
@@ -142,11 +135,31 @@ impl pallet_evm::Config for crate::Runtime {
 	type FeeCalculator = crate::DynamicEvmFee;
 	type FindAuthor = FindAuthorTruncated<Aura>;
 	type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
-	type OnChargeTransaction = TransferEvmFees<DealWithFees>;
+	type OnChargeTransaction = evm_fee::TransferEvmFees<
+		evm_fee::DepositEvmFeeToTreasury,
+		crate::MultiTransactionPayment, // Get account's fee payment asset
+		WethAssetId,
+		ConvertToCurrencyUsingOracle<
+			crate::Runtime,
+			hydradx_adapters::OraclePriceProvider<AssetId, crate::EmaOracle, LRNA>,
+			OracleEvmPeriod,
+		>, // Conversions between assets
+		FungibleCurrencies<crate::Runtime>, // Multi currency support
+	>;
 	type OnCreate = ();
 	type PrecompilesType = precompiles::HydraDXPrecompiles<Self>;
 	type PrecompilesValue = PrecompilesValue;
-	type Runner = pallet_evm::runner::stack::Runner<Self>;
+	type Runner = WrapRunner<
+		Self,
+		pallet_evm::runner::stack::Runner<Self>, // Evm runner that we wrap
+		hydradx_adapters::price::FeeAssetBalanceInCurrencyConvertor<
```

Comment:
another convertor ? why you cannot reuse the same as in `OnChargeTransaction`

Reply:
agree, i can simplify and reuse.